Python/optimizer.c

```diff
@@ -182,6 +183,12 @@ _PyOptimizer_Optimize(
     if (err <= 0) {
         return err;
     }
+
+    if (++executors_created >= 10) {
```

Comment:
Maybe `#define` a useful name for `10` here, like `JIT_CLEANUP_THRESHOLD` or something. You can stick it in `pycore_optimizer.h`.